doc/releases/changelog-0.39.0.md

```diff
@@ -4,227 +4,453 @@
+  be done by specifying a `stopping_condition` in `qml.device.preprocess.decompose`. With `qml.transforms.decompose`, 
+  the user-interface for specifying decompositions is much simpler and more versatile.
+  
+  Decomposing gates in a circuit can be done a few ways:
+
+  * Specifying a `gate_set` comprising PennyLane `Operator`s to decompose into:
+
+    ```python
+    from functools import partial
+
+    dev = qml.device('default.qubit')
+    allowed_gates = {qml.Toffoli, qml.RX, qml.RZ}
+
+    @partial(qml.transforms.decompose, gate_set=allowed_gates)
+    @qml.qnode(dev)
+    def circuit():
+        qml.Hadamard(wires=[0])
+        qml.Toffoli(wires=[0, 1, 2])
+        return qml.expval(qml.Z(0))
+    ```
+
+    ```pycon
+    >>> print(qml.draw(circuit)())
+    0: ──RZ(1.57)──RX(1.57)──RZ(1.57)─╭●─┤  <Z>
+    1: ───────────────────────────────├●─┤
+    2: ───────────────────────────────╰X─┤
+    ```
+  
+  * Specifying a `gate_set` that is defined by a rule (Boolean function). For example, one can specify 
+    an arbitrary gate set to decompose into, so long as the resulting gates only act on one or two qubits:
+
+    ```python
+    @partial(qml.transforms.decompose, gate_set = lambda op: len(op.wires) <= 2)
+    @qml.qnode(dev)
+    def circuit():
+        qml.Toffoli(wires=[0, 1, 2])
+        return qml.expval(qml.Z(0))
+    ```
+
+    ```pycon
+    >>> print(qml.draw(circuit)())
+    0: ───────────╭●───────────╭●────╭●──T──╭●─┤  <Z>
+    1: ────╭●─────│─────╭●─────│───T─╰X──T†─╰X─┤     
+    2: ──H─╰X──T†─╰X──T─╰X──T†─╰X──T──H────────┤
+    ```
+  
+  * Specifying a value for `max_expansion`. By default, decomposition occurs recursively until the desired 
+    gate set is reached, but this can be overridden to control the number of passes. 
+
+    ```python
+    phase = 1.0
+    target_wires = [0]
+    unitary = qml.RX(phase, wires=0).matrix()
+    n_estimation_wires = 1
+    estimation_wires = range(1, n_estimation_wires + 1)
+
+    def qfunc():
+        qml.QuantumPhaseEstimation(
+            unitary,
+            target_wires=target_wires,
+            estimation_wires=estimation_wires,
+        )
+
+    decompose_once = qml.transforms.decompose(qfunc, max_expansion=1)
+    decompose_twice = qml.transforms.decompose(qfunc, max_expansion=2)
+    ```
+
+    ```pycon
+    >>> print(qml.draw(decompose_once)())
+    0: ────╭U(M0)¹───────┤  
+    1: ──H─╰●───────QFT†─┤  
+    M0 = 
+    [[0.87758256+0.j         0.        -0.47942554j]
+    [0.        -0.47942554j 0.87758256+0.j        ]]
+    >>> print(qml.draw(decompose_twice)())
+    0: ──RZ(1.57)──RY(0.50)─╭X──RY(-0.50)──RZ(-6.28)─╭X──RZ(4.71)─┤  
+    1: ──H──────────────────╰●───────────────────────╰●──H†───────┤ 
+    ```
 
-<h4>Calculating Polynomials 🔢</h4>
+<h3>Improvements 🛠</h3>
 
-* `qml.OutPoly` template is added to implement polynomial arithmetic.
-  [(#6320)](https://github.com/PennyLaneAI/pennylane/pull/6320)
+<h4>qjit/jit compatibility and improvements</h4>
```

Comment:
@josh146 re: [your comment](https://github.com/PennyLaneAI/pennylane/pull/6457#discussion_r1821521510)

Does this blurb suffice? Taken from the "What is qjit" page